src/FakeItEasy/Creation/CastleDynamicProxy/CastleDynamicProxyGenerator.cs

```diff
@@ -67,14 +89,11 @@ public ProxyGeneratorResult GenerateProxy(Type typeOfProxy, IEnumerable<Type> ad
                 return new ProxyGeneratorResult(DynamicProxyResources.ProxyIsSealedTypeMessage.FormatInvariant(typeOfProxy));
             }
 
+            Guard.AgainstNull(additionalInterfacesToImplement, "additionalInterfacesToImplement");
             GuardAgainstConstructorArgumentsForInterfaceType(typeOfProxy, argumentsForConstructor);
+            Guard.AgainstNull(fakeCallProcessorProvider, "fakeCallProcessorProvider");

```

Comment:
Having guard clauses for arguments half way through methods is also a massive code smell.